lib/adapters/call-hierarchy-adapter.ts

```diff
@@ -0,0 +1,145 @@
+  async getCallHierarchy<T extends atomIde.CallHierarchyType>(
+    connection: LanguageClientConnection,
+    editor: TextEditor,
+    point: Point,
+    type: T
+  ): Promise<atomIde.CallHierarchy<T>> {
+    const results = await Utils.doWithCancellationToken(connection, this._cancellationTokens, (cancellationToken) =>
+      connection.prepareCallHierarchy(
+        {
+          textDocument: Convert.editorToTextDocumentIdentifier(editor),
+          position: Convert.pointToPosition(point),
+        },
+        cancellationToken
+      )
+    )
+    return <CallHierarchyForAdapter<T>>{
+      type,
+      data: results?.map(convertCallHierarchyItem) ?? [],
+      itemAt(n: number) {
+        if (type === "incoming") {
+          return <Promise<atomIde.CallHierarchy<T>>>this.adapter.getIncoming(this.connection, this.data[n].rawData)
+        } else {
+          return <Promise<atomIde.CallHierarchy<T>>>this.adapter.getOutgoing(this.connection, this.data[n].rawData)
+        }
+      },
+      connection,
+      adapter: this,
+    }
+  }
+  /** Corresponds to lsp's CallHierarchyIncomingCallsRequest. */
+  async getIncoming(
+    connection: LanguageClientConnection,
+    item: CallHierarchyItem
+  ): Promise<atomIde.CallHierarchy<"incoming">> {
+    const results = await Utils.doWithCancellationToken(connection, this._cancellationTokens, (_cancellationToken) =>
+      connection.callHierarchyIncomingCalls({ item })
+    )
+    return <CallHierarchyForAdapter<"incoming">>{
+      type: "incoming",
+      data: results?.map?.((l) => convertCallHierarchyItem(l.from)) || [],
+      itemAt(n: number) {
+        return this.adapter.getIncoming(this.connection, this.data[n].rawData)
+      },
+      connection,
+      adapter: this,
+    }
+  }
+  /** Corresponds to lsp's CallHierarchyOutgoingCallsRequest. */
+  async getOutgoing(
+    connection: LanguageClientConnection,
+    item: CallHierarchyItem
+  ): Promise<atomIde.CallHierarchy<"outgoing">> {
+    const results = await Utils.doWithCancellationToken(connection, this._cancellationTokens, (_cancellationToken) =>
+      connection.callHierarchyOutgoingCalls({ item })
+    )
+    return <CallHierarchyForAdapter<"outgoing">>{
+      type: "outgoing",
+      data: results?.map((l) => convertCallHierarchyItem(l.to)) || [],
+      itemAt(n: number) {
+        return this.adapter.getOutgoing(this.connection, this.data[n].rawData)
+      },
+      connection,
+      adapter: this,
+    }
+  }
+}
+
+function convertCallHierarchyItem(rawData: CallHierarchyItem): CallHierarchyItemForAdapter {
+  return {
+    path: Convert.uriToPath(rawData.uri),
+    name: rawData.name,
+    icon: OutlineViewAdapter.symbolKindToEntityKind(rawData.kind) ?? undefined,
+    tags: rawData.tags
+      ? [
+          ...rawData.tags.reduce((set, tag) => {
+            // filter out null and remove duplicates
+            const entity = symbolTagToEntityKind(tag)
+            return entity == null ? set : set.add(entity)
```

Comment:
```suggestion
            return entity === null ? set : set.add(entity)
```

Reply:
This is what I wanted to do here.
```js
if (entity! == null) {
   set.add (entity)
}
return set
```

The return value of `set.add ()` is `set`, so the original code is probably okay.
Changed `==` to `===`.👍